core/src/main/scala/io/finch/request/package.scala

```diff
@@ -49,14 +129,22 @@ package object request {
     def apply[Req](req: Req)(implicit ev: Req => HttpRequest): Future[A]
 
     def flatMap[B](fn: A => RequestReader[B]) = new RequestReader[B] {
+      val item = MultipleItems
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) = self(req) flatMap { fn(_)(req) }
     }
 
     def map[B](fn: A => B) = new RequestReader[B] {
+      val item = self.item
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) = self(req) map fn
     }
     
+    def mapFuture[B](fn: A => Future[B]) = new RequestReader[B] {
```

Comment:
I have to read the linked article. It's interesting that this method name stirs some debate, as I indeed did not like it much and only picked it due to lack of better ideas.

I'm not sure which name is the most intuitive for people who did not have category theory in primary. :-)